File: src/brs/db/store/SubscriptionStore.java
package brs.db.store;

import brs.Subscription;
import brs.db.SignumKey;
import brs.db.VersionedEntityTable;

import java.util.Collection;

public interface SubscriptionStore {

  SignumKey.LongKeyFactory<Subscription> getSubscriptionDbKeyFactory();

  VersionedEntityTable<Subscription> getSubscriptionTable();

  void saveSubscriptions(Collection<Subscription> subscriptions);

  Collection<Subscription> getSubscriptionsByParticipant(Long accountId);

  Collection<Subscription> getIdSubscriptions(Long accountId);

  Collection<Subscription> getSubscriptionsToId(Long accountId);

  Collection<Subscription> getUpdateSubscriptions(int timestamp);
}

File: src/main/java/brs/db/store/SubscriptionStore.kt
package brs.db.store

import brs.Subscription
import brs.db.BurstKey
import brs.db.VersionedEntityTable

interface SubscriptionStore {
    val subscriptionDbKeyFactory: BurstKey.LongKeyFactory<Subscription>

    val subscriptionTable: VersionedEntityTable<Subscription>

    suspend fun getSubscriptionsByParticipant(accountId: Long?): Collection<Subscription>

    suspend fun getIdSubscriptions(accountId: Long?): Collection<Subscription>

    suspend fun getSubscriptionsToId(accountId: Long?): Collection<Subscription>

    suspend fun getUpdateSubscriptions(timestamp: Int): Collection<Subscription>
}
